test/jasmine/tests/hover_label_test.js

```diff
@@ -1187,6 +1187,44 @@ describe('Test hover label custom styling:', function() {
                 path: ['rgb(68, 68, 68)', 'rgb(255, 255, 255)'],
                 text: [13, 'Arial', 'rgb(255, 255, 255)']
             });
+
+            // test insufficient arrayOk case
+            return Plotly.restyle(gd, 'hoverinfo', [['none']]);
+        })
+        .then(function() {
+            expect(gd.calcdata[0].map(function(o) { return o.hi; })).toEqual(
+                ['none', 'x+y+z+text', 'x+y+z+text'],
```

Comment:
Before this commit, we had

```js
gd.calcdata[0].map(o => o.hi)
// => // ['none']
```

and no fallbacks for `hoverinfo` in the hover routine. Since, the `hoverinfo` default value isn't trivial, I chose add a (another :expressionless: ) `Lib` function `fillArray`.

`Lib.fillArray` is very similar to `Lib.mergeArray`, but instead of looping over the min of `cd[i].length` and `hoverinfo.length`, it loops over `cd[i].length` allowing calcdata items to be filled in with defaults.